src/libraries/System.Security.Cryptography.Cng/tests/CngKeyTests.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Xunit;
+using Xunit.Sdk;
+
+namespace System.Security.Cryptography.Tests
+{
+    public static class CngTests
+    {
+        [Theory]
+        [MemberData(nameof(AllPublicProperties))]
+        public static void AllProperties_ThrowOnDisposal(string propertyName)
+        {
+            // Create a dummy key. Use a fast-ish algorithm.
+            // If we can't query the property even on a fresh key, exclude it from the tests.
+
+            PropertyInfo propInfo = typeof(CngKey).GetProperty(propertyName);
+            CngKey theKey = CngKey.Create(CngAlgorithm.ECDsaP256);
+
+            try
+            {
+                propInfo.GetValue(theKey);
+            }
+            catch
+            {
+                throw new SkipException($"Property getter CngKey.{propertyName} threw an exception; skipping test.");
```

Comment:
SkipException only works if the `Theory` attribute is `[ConditionalTheory]`. Otherwise the test will just throw the exception and fail.

Why would we want to skip the test though?

Reply:
Because if the property setter doesn't work for some reason unrelated to disposal, that's not really in the purview of this test class.